cdap-app-fabric/src/main/java/io/cdap/cdap/internal/provision/NativeProvisioner.java

```diff
@@ -75,14 +75,14 @@ public Cluster createCluster(ProvisionerContext context) {
   }
 
   @Override
-  public ClusterStatus getClusterStatus(ProvisionerContext context, Cluster cluster) {
+  public Cluster getClusterStatus(ProvisionerContext context, Cluster cluster) {
     ClusterStatus status = cluster.getStatus();
-    return status == ClusterStatus.DELETING ? ClusterStatus.NOT_EXISTS : status;
+    return new Cluster(cluster, status == ClusterStatus.DELETING ? ClusterStatus.NOT_EXISTS : status);
```

Comment:
This looks pretty non-obvious and now method name does not does what it means.

Reply:
Removed in the new changes.